radiomesh/gridding.py

```diff
@@ -270,20 +266,22 @@ def apply_w_impl(
           y_kernel = eval_es_kernel(KERNEL, v_grid, v_pixel_start)
           z_kernel = eval_es_kernel(KERNEL, w_grid, w_pixel_start)
 
-          for zfi, zkw in zip(
-            numba.literal_unroll(z_indices), numba.literal_unroll(z_kernel)
-          ):
-            zi = int(zfi)
-            for xfi, xkw in zip(
-              numba.literal_unroll(x_indices), numba.literal_unroll(x_kernel)
-            ):
-              xi = int(xfi)
-              for yfi, ykw in zip(
-                numba.literal_unroll(y_indices), numba.literal_unroll(y_kernel)
-              ):
-                yi = int(yfi)
-                weighted_stokes = apply_weights(vis, xkw * ykw * zkw)
-                accumulate_data(weighted_stokes, vis_grid, (zi, xi, yi), 0)
+          for d in range(ndir):
+            didx = idx + (d,)
+            dir_vis = maybe_apply_jones(JONES_VIS_DATUM, jones_params, vis, didx)
+            dir_weight = maybe_apply_jones(JONES_WGT_DATUM, jones_params, wgt, didx)
+            dir_vis = apply_weights(dir_vis, dir_weight)
+
+            # Conjugation of visibilities must occur *after*
+            # application of jones matrices
+            if should_conjugate:
+              dir_vis = conj_vis(dir_vis)
```

Comment:
@landmanbester So the conjugation of the visibilities in `wgrid_data` occurs after the application of the jones matrices has happened. I wonder if this is strictly necessary and could be happen before.

This would allow us to perform vis conjugation outside the direction loop. This probably isn't a big deal but it might be nicer from a readability perspective.

Reply:
It could happen before. The conjugation should be outside the loop